papergres.go

```diff
@@ -436,89 +447,90 @@ func (s *Schema) InsertAll(objs interface{}) ([]*Result, error) {
 			return
 		}).Exec()
 }
 
+// GenerateInsert generates the insert query for the given object
+func (s *Schema) GenerateInsert(obj interface{}) *Query {
+	return s.generateInsertQuery(obj, false)
+}
+
+// GenerateInsertWithPK generates the insert query for the given object in which
+// PrimaryKey value is also supposed to be populated during insert.
+func (s *Schema) GenerateInsertWithPK(obj interface{}) *Query {
+	return s.generateInsertQuery(obj, true)
+}
+
+// generateInsertQuery constructs an insert query for the given object
+func (s *Schema) generateInsertQuery(obj interface{}, withPK bool) *Query {
+	sql := insertSQL(obj, s.Name, withPK)
+	args := insertArgs(obj, withPK)
+	q := s.Database.Query(sql, args...)
+	q.insert = true
+	return q
+}
+
 // insertSQL generates insert SQL string for a given object and schema
-func insertSQL(obj interface{}, schema string) string {
+func insertSQL(obj interface{}, schema string, withPK bool) string {
+	// Construct the table name prefixed with schema name
 	tname := goToSQLName(getTypeName(obj))
 	tname = fmt.Sprintf("%s.%s", schema, tname)
+
+	// Construct the first component of insert statement
 	sql := fmt.Sprintf("INSERT INTO %s (", tname)
 
-	fields, primary := prepareFields(obj)
+	// NOTE: An object is represented as a slice of Fields, where each Field
+	// represents a column.
+	// Get list of columns to populate.
+	fields, primary := prepareFields(obj, withPK)
+
+	// Based on the number of columns, create value placeholders
 	var values string
 	for i, f := range fields {
 		sql += fmt.Sprintf("\n\t%s,", getColumnName(f))
 		values += fmt.Sprintf("\n\t$%v,", i+1)
 	}
 
+	// Add source data placeholders
+	// something like this: `VALUES ($1, $2, $3, $4, $5, $6)`
 	sql = strings.TrimRight(sql, ",")
 	sql += "\n)\nVALUES ("
 	sql += values
 	sql = strings.TrimRight(sql, ",")
 	sql += "\n)\n"
+
+	// Add last line to capture primary key
 	sql += fmt.Sprintf("RETURNING %s as LastInsertId;", getColumnName(primary))
 
 	return sql
 }
 
-func insertMultipleSQL(entities interface{}, schema string) (string, []interface{}) {
-	slice, err := convertToSlice(entities)
-	if err != nil {
-		return "", nil
-	}
-
-	tname := goToSQLName(getTypeName(slice[0]))
-	tname = fmt.Sprintf("%s.%s", schema, tname)
-	sql := fmt.Sprintf("INSERT INTO %s (", tname)
-
-	fields, _ := prepareFields(slice[0])
-	for _, f := range fields {
-		sql += fmt.Sprintf("\n\t%s,", getColumnName(f))
-	}
-	sql = strings.TrimRight(sql, ",")
-	sql += "\n)\nVALUES \n"
-
-	args := make([]interface{}, len(slice)*len(fields))
-
-	counter := 0
-	for _, e := range slice {
-		fs, _ := prepareFields(e)
-		sql += fmt.Sprintf("(")
-		for _, f := range fs {
-			sql += fmt.Sprintf("$%v, ", counter+1)
-			args[counter] = f.Value
-			counter++
-		}
-		sql = strings.TrimRight(sql, ", ")
-		sql += fmt.Sprintf("),\n")
-	}
-	sql = strings.TrimRight(sql, ",\n")
-
-	return sql, args
-}
-
+// getColumnName returns a Field's associated Tag name if it is supplied.
```

Comment:
See above regarding comments

Reply:
Same as above. Line comments are the norm.